src/org/opendatakit/briefcase/ui/automation/AutomationPanelForm.java

```diff
@@ -0,0 +1,144 @@
+package org.opendatakit.briefcase.ui.automation;
+
+import static org.opendatakit.briefcase.ui.reused.FileChooser.directory;
+
+import java.awt.GridBagConstraints;
+import java.awt.GridBagLayout;
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Optional;
+import java.util.function.Consumer;
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JLabel;
+import javax.swing.JPanel;
+import javax.swing.JTextField;
+import org.opendatakit.briefcase.automation.AutomationConfiguration;
+import org.opendatakit.briefcase.util.StringUtils;
+
+@SuppressWarnings("checkstyle:MethodName")
+public class AutomationPanelForm {
+  public JPanel container;
+  private JLabel scriptDirLabel;
+  private JTextField scriptDirField;
+  private JButton scriptDirChooseButton;
+  private JButton generateScriptButton;
+  private JButton changeDefaultConfigurationButton;
+
+  public AutomationPanelForm() {
+    $$$setupUI$$$();
+    scriptDirChooseButton.addActionListener(__ ->
+        directory(container, fileFrom(scriptDirField))
+            .choose()
+            .ifPresent(file -> setScriptDir(Paths.get(file.toURI()))));
+  }
+
+  void onSetExportConfiguration(Runnable runnable) {
+    changeDefaultConfigurationButton.addActionListener(__ -> runnable.run());
+  }
+
+  void onGenerate(Consumer<AutomationConfiguration> callback) {
+    AutomationConfiguration config = AutomationConfiguration.empty();
```

Comment:
The creation of the config object should happen inside the listener's block. Here it will be created just once, when `AutomationPanel` calls this method and won't get the changes from the user's interaction with the UI.

Reply:
Ok